### ISSUE TITLE
automl / add more choices of optimization_objective in automl_config

### ISSUE THREAD

**Issue by gionatamettifogo** [analitico/analitico-automl#8]
When building estimator we should fix the loss function/optimization objective from the configuration, eg:

class OptimizationObjective(Enum):
    """
    Objective function the model is optimizing towards. The training process creates a model
    that maximizes/minimizes the value of the objective function over the validation set.
    The supported optimization objectives depend on the prediction type. If the field is not set,
    a default objective function is used.
    """

    # Maximize the area under the receiver operating characteristic (ROC) curve (CLASSIFICATION_BINARY default)
    MAXIMIZE_AU_ROC = "maximize-au-roc"
    # Minimize log loss (CLASSIFICATION_BINARY | CLASSIFICATION_MULTI_CLASS default)
    MINIMIZE_LOG_LOSS = "minimize-log-loss"
    # Maximize the area under the precision-recall curve (CLASSIFICATION_BINARY)
    MAXIMIZE_AU_PRC = "maximize-au-prc"
    # Minimize root-mean-squared error (RMSE) (REGRESSION default)
    MINIMIZE_RMSE = "minimize-rmse"
    # Minimize mean-absolute error (MAE) (REGRESSION)
    MINIMIZE_MAE = "minimize-mae"
    # Minimize root-mean-squared log error (RMSLE) (REGRESSION)
    MINIMIZE_RMSLE = "minimize-rmsle"

**Comment by fmardero** [comment]
First try:   #19

**Comment by gionatamettifogo** [comment]
Looks good, pulled the code. Let's add the other loss functions next.

**Comment by fmardero** [comment]
Multi-class classification now should be good. Every prediction task works fine for me. Added new regression loss functions.

In classification tasks the log-loss function is almost always used. The other optimization objectives are not directly differentiable (i.g. auc-roc minimization): I need to find a proxy to this metrics.

**Comment by fmardero** [comment]
Missing for classification:
* **MAXIMIZE_AU_PRC** - AUC PR: The area under the precision-recall (PR) curve
* **MAXIMIZE_AU_ROC** - AUC ROC: The area under the receiver operating characteristic (ROC) curve
* **MAXIMIZE_PRECISION_AT_RECALL** - Optimize precision at a specific recall value (while optimizing all the network)
* **MAXIMIZE_RECALL_AT_PRECISION** - Optimize recall at a specific precision value (while optimizing all the network)
* **MINIMIZE_FOCAL LOSS** [my contribution] - very very useful loss when dealing with imbalanced datasets

For the last one I'm waiting for the `tensorflow_addon` library to be included in the repo.

**Comment by fmardero** [comment]
Added focal loss for binary 2f3c5d4f4684636bd3c7e9494c1179a716747580 and multi-class b13c6a2acb3913daf4dfe89d11cb3813fea28f34 tasks.

**Comment by gionatamettifogo** [comment]
Back to the original objective of implementing these:

# Maximize the area under the receiver operating characteristic (ROC) curve (CLASSIFICATION_BINARY default)
    # Distinguish between classes. Default value for binary classification.
    MAXIMIZE_AU_ROC = "maximize-au-roc"
    # Minimize log loss (CLASSIFICATION_BINARY | CLASSIFICATION_MULTI_CLASS default)
    # Keep prediction probabilities as accurate as possible. Default value for multi-class classification.
    MINIMIZE_LOG_LOSS = "minimize-log-loss"
    # Maximize the area under the precision-recall curve (CLASSIFICATION_BINARY)
    # Optimize results for predictions for the less common class.
    MAXIMIZE_AU_PRC = "maximize-au-prc"
    # Minimize root-mean-squared error (RMSE) (REGRESSION default)
    # Capture more extreme values accurately.
    MINIMIZE_RMSE = "minimize-rmse"
    # Minimize mean-absolute error (MAE) (REGRESSION)
    # View extreme values as outliers with less impact on model.
    MINIMIZE_MAE = "minimize-mae"
    # Minimize root-mean-squared log error (RMSLE) (REGRESSION)
    # Penalize error on relative size rather than absolute value. Especially helpful when both predicted and actual values can be quite large.
    MINIMIZE_RMSLE = "minimize-rmsle"

I moved the related code to /utilities/loss.py
I created a unit test file for the different loss functions with titanic, boston, iris as a started
Let's add the missing objectives (or use focal loss for now if we can't get them to work)
